js_modules/dagster-ui/packages/ui-core/src/workspace/WorkspaceContext.tsx

```diff
@@ -186,9 +192,17 @@ export const ROOT_WORKSPACE_QUERY = gql`
  * in the workspace, and loading/error state for the relevant query.
  */
 const useWorkspaceState = (): WorkspaceState => {
-  const {data, loading, refetch} = useQuery<RootWorkspaceQuery, RootWorkspaceQueryVariables>(
-    ROOT_WORKSPACE_QUERY,
-  );
+  const {
+    data,
+    loading,
+    fetch: refetch,
+  } = useIndexedDBCachedQuery<RootWorkspaceQuery, RootWorkspaceQueryVariables>({
```

Comment:
This is great!  I'm a big fan but we should verify this doesn't break anything in scenarios where the root workspace clearing on a full page load might have been good.

eg: switching orgs in cloud --  since the cache key is constant, maybe when you open one org you will now see the other org's root workspace info for a minute?

Reply:
Our orgs all use subdomains which should isolate them, but it is actually possible to access ur org without a subdomain so yeah I'll add the org name as part of the key for now

Reply:
I'll do this in a follow up pr, I think we just want to do this generally in the app across localStorage and indexedDB both.